sound/soc/sof/loader.c

```diff
@@ -338,6 +338,12 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 		return ret;
 	}
 
+	/*
+	 * At this point, fw boot is complete and only core 0 is active.
+	 * So, update the active cores mask.
```

Comment:
How generic a solution is this? Wondering if there isn't an Intel-specific assumption that gets pushed into the generic layers. There are other solutions with multi-cores and hard-coding that only core0 is active after boot seems like a very strong assumption?

Reply:
@plbossart the FW has something called a MASTER_CORE_ID defined per platform. Maybe we could use that instead?

Reply:
@ranj063 indeed, we could probably extend the description of the cores below and add this notion of "core available on startup"

````c
const struct sof_intel_dsp_desc apl_chip_info = {
	/* Apollolake */
	.cores_num = 2,
	.cores_mask = HDA_DSP_CORE_MASK(0) | HDA_DSP_CORE_MASK(1),
````

Reply:
@plbossart updated this PR to add the master core id to chip info.